DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/svd_testcase.hpp

```diff
@@ -0,0 +1,169 @@
+#ifndef __SVD_TESTCASE__
+#define __SVD_TESTCASE__
+
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <vector>
+
+#include "svd.hpp"
+#include "svd_testbench_tool.hpp"
+#include "print_matrix.hpp"
+
+template <typename T, unsigned rows_A, unsigned cols_A, bool is_complex = false>
```

Comment:
I like this test framework you built, but can you comment it a little bit to explain how it works? I could see wanting to use a a framework like this in other code samples too as it makes your `main()` function really tidy! 😊